### ISSUE TITLE
More code ligature customization

### ISSUE THREAD

**Issue by thesecretmaster** [arrowtype/recursive#493]
**Is your feature request related to a problem? Please describe.**

Would it be possible to allow further customization (either by scripting or by some other means) of which code ligatures are enabled in recursive? I've been reading various resources on how different coding fonts have approached coding ligatures, and specifically a lot of stuff @/arrowtype has written. Similar to the idea expressed [here](https://github.com/tonsky/FiraCode/issues/854), I believe that coding ligatures can be split into two groups, those that are fairly intuitive and simply alter some characters to flow better together (i.e. =>, ==, ||, >>=, /=) and those that are a little less intuitive and represent the same semantic meaning, but look different visually (i.e. !=, <=, >=) (which I've been calling "semantic ligatures", because they preserve semantic meaning but don't represent the underlying characters as much). I dislike these "semantic ligatures" and I've been looking for ways to turn them off while preserving the rest of the ligatures.

**Describe the solution you'd like**

There are multiple solutions to this issue, most of which are outlined already in [docs/code-ligatures--implementation](https://github.com/arrowtype/recursive/tree/main/docs/code-ligatures--implementation), so I won't go over them again since it seems like you're well aware of them. Assuming that you won't change the decision that was made there (to use dlig), I'd like some instruction on how to manually remove ligatures that I don't like during the build process.

**Describe what you've tried so far**

I've tried altering `src/features/features/dlig-generated.fea` and `src/features/features/calt-generated--code_fonts_only.fea` to remove the ligatures I dislike, but it appears that they are still in the generated font. I assume `dlig-generated` is a generated file but it's not overwritten in the build process and I can't figure out what generates it, or why removing ligatures there does not remove them in the produced font files.

**Additional context**

> Side note: My personal feeling is that the perfect approach would be to enable the more intuitive ligatures via calt (with the rationale that those simply will read better in almost any context) and to put the less intuitive "semantic ligatures" in dlig as those ligatures make more substantial changes. This is similar to the CODE axis idea described in docs/code-ligatures--implementation, but with only 3 steps.

**Comment by thesecretmaster** [comment]
Ligatures are overwritten in [recursive-code-config](https://github.com/arrowtype/recursive-code-config), so you can change the ligatures there. The changes I make _are_ built in to the font output files, but my terminal (kitty) doesn't seem to support them correctly, which is why I was confused.

**Comment by arrowtype** [comment]
@thesecretmaster Sorry for my delayed response here. Thanks for taking time to file an issue! 

The fonts produced by the recursive-code-confit repo should work in most scenarios... if you have time, would you mind sharing how they are failing for you?

**Comment by arrowtype** [comment]
It's been awhile since I have implemented the ligatures, so I would need to dig back in to answer your questions around customizing specific ligatures to exclude. I can see how that would be a beneficial feature, though.

**Comment by thesecretmaster** [comment]
Yeah, the fonts output by recursive-code-config do work, the reason I was confused is because I was building with the following steps:

1. Remove ligatures in the `recursive` repository by editing `dlig-generated` and `calt...generated`
2. Build `recursive` repository
3. Try to use recursive (the variable ttf file) in vim in my terminal (kitty) -- no ligatures appear
4. Copy variable ttf file into `recursive-code-config`
5. Build recursive code config (which, as I learned about 10 minutes ago, uses the features in **its** feature directory to overwrite the features in the output font)
6. Try to use the built fonts, see that even the ligatures I tried to remove are showing up, very confusing

But now I've figured out that if I edit the feature files in `recursive-code-config` I can produce the fonts I need, so my direct problem is solved. It would be cool to have a more explicit way to do this, i.e. with font features but it's not exactly a problem as is, so I'm not sure if it's important or not.

> Side note: I do think that I've run into an actual issue with either recursive or my terminal emulator (kitty), since with the fonts output by `recursive` (not `recursive-code-config`) even if I enable `dlig` in kitty, I don't get the coding ligatures. But I'm unsure if that's just kitty not supporting variable ttf or if it's something about the font. Either way, that's a separate issue and if I can't figure it out I'll probably make a separate GH issue for it. Currently I'm thinking it's probably kittys fault based on [this issue](https://github.com/kovidgoyal/kitty/issues/297), but I'm still digging.
